client/src/components/viewers/FileControlsInput.vue

```diff
@@ -0,0 +1,147 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <span
+    v-if="!editing"
+    class="file-controls-input text-only"
+    @click="onTextClick"
+  >
+    {{ modelValue }}
+    {{ props.suffix }}
+  </span>
+  <ion-input
+    v-else
+    class="file-controls-input editing-input"
+    fill="outline"
+    ref="inputRef"
+    :value="modelValue"
+    @ion-blur="onFocusChanged(false)"
+    @ion-focus="onFocusChanged(true)"
+    @ion-input="onChange($event.detail.value || '')"
+    @keyup.enter="enterPressed($event.target.value)"
+    @keyup.esc="onFocusChanged(false)"
+    :disabled="$props.disabled"
+    v-bind="$attrs"
+  >
+    <span
+      class="file-controls-input-suffix"
+      slot="end"
+    >
+      {{ props.suffix }}
+    </span>
+  </ion-input>
+</template>
+
+<script setup lang="ts">
+import { IonInput } from '@ionic/vue';
+import { ref } from 'vue';
+
+const editing = ref(false);
+const inputRef = ref();
+const lostFocus = ref(false);
+
+const props = defineProps<{
+  modelValue?: string;
+  disabled?: boolean;
+  restrictChange?: (value: string) => Promise<string>;
+  suffix?: string;
+}>();
+
+const emits = defineEmits<{
+  (e: 'update:modelValue', value: string): void;
+  (e: 'change', value: string): void;
+  (e: 'onSubmittedValue', value: string): void;
+  (e: 'onEnterKeyup', value: string): void;
+  (e: 'onFocusChanged', value: boolean): void;
+}>();
+
+defineExpose({
+  setFocus,
+});
+
+function setFocus(): void {
+  setTimeout(() => {
+    if (inputRef.value && inputRef.value.$el) {
+      inputRef.value.$el.setFocus();
+    }
+  }, 200);
+}
+
+async function onFocusChanged(focus: boolean): Promise<void> {
+  if (focus === false) {
+    lostFocus.value = true;
+    editing.value = false;
+    await onSubmittedValue(props.modelValue || '');
+  }
+  emits('onFocusChanged', focus);
+}
+
+async function enterPressed(value: string): Promise<void> {
+  editing.value = false;
+  emits('onEnterKeyup', value);
+  await onSubmittedValue(value);
+}
+
+async function onChange(value: string): Promise<void> {
+  if (props.restrictChange) {
+    const restrictedValue = await props.restrictChange(value);
+    if (restrictedValue) {
```

Comment:
I'm guessing you would need to change the value of the input too. If I enter `a b` but the restriction removes everything after the space, `restrictedValue` would be `a` but the input would still display `a b`. Maybe it does work if the prop `modelValue` is used, but it's an optional prop.

Reply:
for now, I'll just force modelValue as prop